### PR TITLE
Improve requirements

### PR DESCRIPTION
**Pull request by WillemWybo**
Removes requirements from root dir and moves it to `requirements/`

Removes `dill` and `pathos` from requirements. Applies associated changes to `compartmentfitter`. Modifies `sovtree` to not rely on lambda-functions that can only be pickled with `dill`.

### CODE REVIEW

File: neat/trees/compartmenttree.py [hunk 1]
```diff
@@ -341,10 +339,7 @@ def getDynamicDrive(self, channel_name, p_open, v):
     def getDynamicDrive_(self, channel_name, v, dt, channel_storage=None):
         # assert p_open.shape == v.shape
         _, e = self.currents[channel_name]
-        if channel_storage is not None:
```
Comment (jakobj) [inline neat/trees/compartmenttree.py:344]: not sure these changes are on purpose as they don't seem related to fixing reqs?

File: neat/trees/sovtree.py [hunk 2]
```diff
@@ -47,46 +47,47 @@ def _setSOV(self, channel_storage, tau_0=0.02):
         self.tau_0      = tau_0 # s
         self.z_a        = self.r_a / (np.pi * self.R_sov**2) # MOhm/cm
         self.g_inf_m    = 1. / (self.z_a * self.lambda_m) # uS
-        # function for SOV approach
```
Comment (jakobj) [inline neat/trees/sovtree.py:50]: (same question as above)

File: neat/trees/sovtree.py [hunk 3]
```diff
@@ -181,37 +183,33 @@ def _setSOV(self, channel_storage, tau_0=0.02):
         self.g_s        = self.g_m*self.A + self.g_shunt # uS
         self.c_s        = self.c_m*self.A # uF
         self.tau_0      = tau_0  # s
-        # functions for the SOV approach
-        self.f_transc   = lambda x: 0.
-        self.dN_dp      = lambda x: 0.
         # segment amplitude factors
         self.kappa_m = 1.
 
-    def _setMuFunctions(self):
```
Comment (jakobj) [inline neat/trees/sovtree.py:190]: (same question as above)

**Review by jakobj — CHANGES_REQUESTED** [review]
there are a lot of changes i'm not sure about. are they directly related to improving reqs? do you need them bc you remove pickle?

changes in `requirements.txt` look good, i like the version tagging. but what's the intention behind creating a separate directory?

File: neat/trees/sovtree.py (hunk 2 above)
Comment (WillemWybo) [reply neat/trees/sovtree.py:50]: Changes were needed to remove the dependency on `dill`

File: neat/trees/sovtree.py (hunk 3 above)
Comment (WillemWybo) [reply neat/trees/sovtree.py:190]: same as above

File: neat/trees/compartmenttree.py (hunk 1 above)
Comment (WillemWybo) [reply neat/trees/compartmenttree.py:344]: Came across this deprecated part of the code while checking whether some import statements could be removed.

**Comment by WillemWybo** [comment]
@jakobj yes the changes are needed to remove dependency on dill, as with standard pickle we can not use lambda functions. @benelot implemented the separate directory. I presume he wanted a separate file for the doc requirements, but still group all requirements together in a directory

**Comment by WillemWybo** [comment]
@benelot currently the file `requirements/doc.txt` still contains all dependencies of the main toolbox: 
```
# documentation
sphinx>=2.4.3
nbsphinx>=0.5.1
sphinx-gallery>=0.6.1
sphinx_rtd_theme>=0.4.3
nb2plots>=0.6
pillow>=7.1.2

numpy>=1.14.1
matplotlib>=2.1.2
cython>=0.27.3
scipy>=1.0.0
sympy>=1.1.1
dill>=0.3.1.1
pathos>=0.2.5
pytest>=4.3.0
scikit_learn>=0.22.2.post1
```
Can the second part (after the empty line) be removed?

**Comment by benelot** [comment]
Exactly, I separated them in order to not install all of them just for the documentation part.

**Comment by benelot** [comment]
I think we can remove the second part from the docs requirements. I mean the following requirements

```
numpy>=1.14.1
matplotlib>=2.1.2
cython>=0.27.3
scipy>=1.0.0
sympy>=1.1.1
dill>=0.3.1.1
pathos>=0.2.5
pytest>=4.3.0
scikit_learn>=0.22.2.post1
```
The docs should run through without them. In case it does not, I can fix it afterwards again by checking why it breaks on the docs side. For the release, this will not matter.

**Review by benelot — APPROVED** [review]
This is exactly how I would do the requirements.txt handling. The separate folder is not necessary, but reduces the clutter of the repository base folder. From what I see in other repositories, there are still a lot of standard documents that will reside in the base folder after some years, so it is good to clean up from time to time.
